pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -108,6 +127,15 @@ func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error {
 		IsController: true,
 		OwnerType:    &hivev1.ClusterDeployment{},
 	})
+
+	// Watch for pods created by a ClusterDeployment:
+	/*err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &hivev1.ClusterDeployment{},
+	})*/
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(reconciler.selectorPodWatchHandler),
+	})
```

Comment:
Not sure if you worked this out with Cesar or not, my past experience with this was: https://github.com/openshift/cloud-credential-operator/blob/master/pkg/controller/credentialsrequest/credentialsrequest_controller.go#L119-L176

I think the request func should be removed from the Reconciler object and just be a regular func in this module, or inline like the link. 

I'm not sure if you need a predicate or not, more comments on the predicate in the handler below.